### DISCUSSION TITLE
PM only reads 1 of my lines of error coefficients; 2 outputs in model

### DISCUSSION THREAD

**Discussion by livPKPD** [LAPKB/Pmetrics#66] [General]
Hi there, I have tried writing a model in the console as below, but I get the error: 

_MODEL REPORT:
There are errors in your model file.
There is 1 line of error coefficients in the model file, but 2 output equations.
These must be the same._

Here is my model (I have taken out some elements to make it more readable - the problem seems to be with the 'out' block):
mod1b <- PM_model$new(list(
  pri = list(...),
  sec = list(...),
  cov = list(...),
  dif = list(...),
  out = list(
    Y1 = list(
      value = "X(2)/V",
      err = list(
        model = additive(5),
        assay = c(0.01, 0.1, 0, 0)
      )
    ),
    Y2 = list(
      value = "X(3)/Vcns",
      err = list (
        model = additive(5),     
        assay = c(0.01, 0.1, 0, 0)
      )
    )
  )
))

The model ran fine in the S3 version of PM. But even when I load the .txt file using PM_model$new(), I get the same error code as above.
Can anyone help please?
Thanks!

**Comment by Siel** [comment]
Hi @livPKPD Could you please paste the full model here? so I can try to run it locally.

Thanks

**Reply by livPKPD** [reply]
Hi, yes it's:

mod1b <- PM_model$new(list(
  pri = list(
    Ka = range(0.1,7),
    V = range(0.1,50),
    Kcp = range(0.01,80),
    Kpc = range(0.01,50),
    Kcs = range(0.01,80),
    Ksc = range(0.01,50),
    Vcns = range(0.01, 60),
    SCL = range(0.01,15)
  ),
  cov = list("AGE", "sex_m0_f1", "prevflc", "creat", "alt", "arm", "weight", "creatmgdl", "crcl"),
  dif = list(
    xp1 = "-Ka * X(1)",
    xp2 = "- (Kcp + Kcs + (SCL/V))*X(2) + (Ksc*X(3)) + (Kpc*X(4)) + (Ka*X(1))",
    xp3 = "Kcs*X(2) - Ksc*X(3)",
    xp4 = "Kcp*X(2) - Kpc*X(4)"
  ),
  out = list(
    Y1 = list(
      value = "X(2)/V",
      err = list(
        model = additive(3),
        assay = c(0.025, 0.1, 0, 0)
      )
    ),
    Y2 = list(
      value = "X(3)/Vcns",
      err = list (
        model = additive(3),      
        assay = c(0.25, 0.1, 0, 0)
      )
    )
  )
))

Many thanks for any insight!

**Comment by Siel** [comment]
This is the translated model of that structure
````
#pri
Ka, 0.100000, 7.000000
V, 0.100000, 50.000000
Kcp, 0.010000, 80.000000
Kpc, 0.010000, 50.000000
Kcs, 0.010000, 80.000000
Ksc, 0.010000, 50.000000
Vcns, 0.010000, 60.000000
SCL, 0.010000, 15.000000

#cov
AGE
sex_m0_f1
prevflc
creat
alt
arm
weight
creatmgdl
crcl

#dif
XP(1)=-Ka * X(1)
XP(2)=- (Kcp + Kcs + (SCL/V))X(2) + (KscX(3)) + (KpcX(4)) + (KaX(1))
XP(3)=KcsX(2) - KscX(3)
XP(4)=KcpX(2) - KpcX(4)

#out
Y(1)=X(2)/V
Y(2)=X(3)/Vcns

#err
L=3.000000
0.025000,0.100000,0.000000,0.000000
````

@mnneely do I have to include a second line of coefficients?

**Reply by livPKPD** [reply]
As a follow up - I have run the model in the S3 version of PMetrics; but now when I do my VPC with 
valid<- PM_load() 
valid$validate(limits=NA)

I have 2 issues; the first is that the data validation report gives this error - I'm not sure why, because my data file does have time=0 as the first row for all patients: 
The following row numbers do not have time=0 as first record: See the 'errors.xlsx' file in current directory.... then a long list of line numbers.

The second issue is that the "simmodel.txt" file that is created during the $validate run only contains 1 line of error coefficients - so again, it does not seem to be reading the second line, and again I get:
Error: 
There is 1 line of error coefficients in the model file, but 2 output equations.
These must be the same.

**Reply by livPKPD** [reply]
Hi again - just wondering if you can give me any pointers on this? Many thanks :)

**Comment by mnneely** [comment]
@livPKPD, yes you do need two lines of assay error coefficients. I think you have specified your model correctly, but it  isn't being translated correctly. I will open an issue on this and get it fixed.

**Comment by mnneely** [comment]
I believe I have fixed this bug. Please see Issue here: https://github.com/LAPKB/Pmetrics/issues/68#issuecomment-1161562576.

**Comment by livPKPD** [comment]
Thank you yes that works!!
